qiskit/qpy/binary_io/value.py

```diff
@@ -452,20 +455,27 @@ def _read_parameter(file_obj):
     return Parameter(name, uuid=param_uuid)
 
 
-def _read_parameter_vec(file_obj, vectors):
+def _read_parameter_vec(file_obj, vectors, initialize_full_vec):
     data = formats.PARAMETER_VECTOR_ELEMENT(
         *struct.unpack(
             formats.PARAMETER_VECTOR_ELEMENT_PACK,
             file_obj.read(formats.PARAMETER_VECTOR_ELEMENT_SIZE),
         ),
     )
+    # Starting in version 15, the parameter uuid is used instead of the
+    # parameter name.
     param_uuid = uuid.UUID(bytes=data.uuid)
     name = file_obj.read(data.vector_name_size).decode(common.ENCODE)
-    if name not in vectors:
-        vectors[name] = (ParameterVector(name, data.vector_size), set())
-    vector = vectors[name][0]
-    if vector[data.index].uuid != param_uuid:
-        vectors[name][1].add(data.index)
+
+    if param_uuid not in vectors:
+        vectors[param_uuid] = (ParameterVector(name, data.vector_size), set())
+    vector = vectors[param_uuid][0]
+    if initialize_full_vec:
```

Comment:
Two questions on this:
1. Do we only want to run the initialization if the vector does not exist in the dict yet? (I.e. move this into the if-clause above? 🙂)
2. Would it be safer to just always initialize the full vector? Or are there reasons not to do this?